LaTeX/Parts/Chapter5.tex

```diff
@@ -17,30 +17,53 @@ \section{Varroa Distributed System Architecture}
 Every Agent holds a number of MQTT Agents and each MQTT Agent manages one MQTT Client.
 \newpage
 
-\section{Chunk Distribution}
-To secure a deterministic execution of a scenario the every chunk needs to be distributed to the same Agent in every execution of the scenario.
-To ensure this determinism a mechanism was introduced combining a round-robin selection of chunks with a handshake mechanism of the agents.
+-\section{Chunk Concept}
+A chunk is a work package of the scenario execution designed to be distributed from the Commander to its Agents.
+It is then handled by the Agent it is distributed to.
+The Agent now creates or uses existing MQTT Agents to execute the Chunk.
+It contains:
+\begin{itemize}
+	\item \textbf{clientCountInChunk}: the maximum amount of clients a chunk can represent.
+	\item \textbf{clientOffset}: starting index of the clients contained in the chunk.
+	\item \textbf{clientGroupId}: identfier for the Client Group contain in the chunk.
+	\item \textbf{clientGroupInformation}: contains information concerning the Client Group and MQTT-specific properties.
+	\item \textbf{stageDuration}: the duration of the stage this chunk belongs to.
+	\item \textbf{lifecycleRate}: the rate at which the contained commands are executed.
+	\item \textbf{commands}: a list of commands that are executed by the clients in the chunk.
+\end{itemize}
+%Due to the distributed nature of Varroa it was necessary to be able to spread the information contained in the scenario.
+%To secure a deterministic distribution of a scenario's data the chunks had to be evenly spread across the initialized Agents.
+
+\subsection{Chunk Distribution}
 \begin{figure}[h]
 	\begin{center}
-	\includegraphics[scale=1]{Resources/PDF/ChunkDistribution}
+	\includegraphics[scale=1.2]{Resources/PDF/ChunkDistribution}
 	\caption{Distribution of a Chunk}
 	\label{pic:ChunkDistribution}
 	\end{center}
 \end{figure}
+To secure a deterministic execution of a scenario the every chunk needs to be distributed to the same Agent in every execution of the scenario.
```

Comment:
"every chunk needs to be distributed to the same Agent" is wrong.
```suggestion
A Chunk is deterministically assigned to an Agent to ensure that it gets executed by the same Agent even on multiple runs of the scenario. This determinism is needed to avoid fluctuation of reporting results that would be caused by Varroa and not the MQTT system under test.
```